test/browser/spec/ol/interaction/snap.test.js

```diff
@@ -71,12 +71,19 @@ describe('ol.interaction.Snap', function () {
         coordinate: [0, 0],
         map: map,
       };
+      snapInteraction.on('snap', function (snapEvent) {
+        expect(snapEvent.type).to.be('snap');
+        expect(snapEvent.vertex).to.eql([0, 0]);
+        expect(snapEvent.vertexPixel).to.eql([width / 2, height / 2]);
+        expect(snapEvent.feature).to.eql(point);
+        done();
+      });
       snapInteraction.handleEvent(event);
       // check that the coordinate is in XY and not XYZ
       expect(event.coordinate).to.eql([0, 0]);
```

Comment:
Since events are dispatched synchronously in OpenLayers, this test won't be executed any more.  I'd suggest creating a separate test for the `snap` event instead.

Reply:
Moved event.coordinate `expect` inside `eventHandler` `expect`.
Keeping it in the same unit test for ease of maintenance, as we are doing snapEvent.feature `expect`,
(same for all unit tests below)
and reuse event input data.

Is it still better to have separate unit tests for snapEvent  @ahocevar  ? 
If so I'll refactor.